plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md

```diff
@@ -0,0 +1,531 @@
+- [Using the API](#using-the-api)
+    - [Options](#options)
+        - [General options](#general-options)
+        - [Options for `text` fields](#options-for-text-fields)
+        - [Options for `select` fields](#options-for-select-fields)
+        - [Options for `checkbox` fields](#options-for-checkbox-fields)
+    - [Attributes](#attributes)
+- [Usage examples](#usage-examples)
+    - [Rendering a text field](#rendering-a-text-field)
+    - [Rendering a checkbox field](#rendering-a-checkbox-field)
+    - [Rendering a select field](#rendering-a-select-field)
+    - [The select input before being focused](#the-select-input-before-being-focused)
+    - [The select input when focused](#the-select-input-when-focused)
+- [Validation and sanitization](#validation-and-sanitization)
+    - [Sanitization](#sanitization)
+        - [Using the `woocommerce_blocks_sanitize_additional_field` filter](#using-the-woocommerce_blocks_sanitize_additional_field-filter)
+            - [Example of sanitization](#example-of-sanitization)
+    - [Validation](#validation)
+        - [Single field validation](#single-field-validation)
+            - [Using the `woocommerce_blocks_validate_additional_field` action](#using-the-woocommerce_blocks_validate_additional_field-action)
+                - [The `WP_Error` object](#the-wp_error-object)
+                - [Example of single-field validation](#example-of-single-field-validation)
+        - [Multiple field validation](#multiple-field-validation)
+            - [Using the `woocommerce_blocks_validate_location_{location}_fields` action](#using-the-woocommerce_blocks_validate_location_location_fields-action)
+            - [Example of location validation](#example-of-location-validation)
+- [A full example](#a-full-example)
+
+A common use-case for developers and merchants is to add a new field to the Checkout form to collect additional data about a customer or their order.
+
+This document will outline the steps an extension should take to register some additional checkout fields.
+
+## Available field locations
+
+Additional checkout fields can be registered in three different places:
+
+- Contact information
+- Addresses (Shipping **and** Billing)
+- Additional information
+
+A field can only be shown in one location, it is not possible to render the same field in multiple locations in the same registration.
+
+### Contact information
+
+The contact information section currently renders at the top of the form. It contains the `email` field and any other additional fields.
+
+![Showing the contact information section with two fields rendered, email and an additional checkout field (optional)](https://github.com/woocommerce/woocommerce/assets/5656702/097c2596-c629-4eab-9604-577ee7a14cfe)
+
+Fields rendered here will be saved to the _order_ (i.e. they will not be part of the customer's saved address or account information. New orders will not have any previously used values pre-filled).
```

Comment:
We're saving contact details to account. They render under my account > account details.